DisplayBundle/Tests/Manager/CacheableManagerTest.php

```diff
@@ -37,28 +37,31 @@ public function testSetPublicResponseCacheParameters($maxAge, $expectedMaxAge, $
     {
         $response = Phake::mock('Symfony\Component\HttpFoundation\Response');
 
-        $newResponse = $this->manager->setResponseCacheParameters($response, $maxAge, 'public');
+        $newResponse = $this->manager->setResponseCacheParameters($response, $maxAge, 'public', true);
 
         $this->assertSame($response, $newResponse);
         Phake::verify($newResponse)->setPublic();
-        Phake::verify($newResponse, Phake::times($count))->setMaxAge($expectedMaxAge);
+        Phake::verify($newResponse, Phake::times($count-1))->setMaxAge($expectedMaxAge);
+        Phake::verify($newResponse, Phake::times($count))->setSharedMaxAge($expectedMaxAge);
     }
 
     /**
      * @param int $maxAge
      * @param int $expectedMaxAge
+     * @param int $count
      *
      * @dataProvider provideMaxAge
      */
     public function testSetPrivateResponseCacheParameters($maxAge, $expectedMaxAge, $count)
     {
         $response = Phake::mock('Symfony\Component\HttpFoundation\Response');
 
-        $newResponse = $this->manager->setResponseCacheParameters($response, $maxAge, 'private');
+        $newResponse = $this->manager->setResponseCacheParameters($response, $maxAge, 'private', false);
 
         $this->assertSame($response, $newResponse);
         Phake::verify($newResponse)->setPrivate();
         Phake::verify($newResponse, Phake::times($count))->setMaxAge($expectedMaxAge);
+        Phake::verify($newResponse, Phake::times($count-1))->setSharedMaxAge($expectedMaxAge);
     }
 
     /**
```

Comment:
You're right i forget to test public response without  ESI, but i don't think that i have to test private response with ESI because this case is not used. in fact, to take advantage of shared caches the response will need to be explicitly set as public and that what we do in the NodeController